### PR TITLE
build peg-out tx

### PR DESCRIPTION
**Pull request by donpdonp**
copy the working peg-out builder from frost_btc test into stacks-coordinator.

### CODE REVIEW

File: stacks-coordinator/src/peg_wallet.rs [hunk 1]
```diff
@@ -97,15 +98,21 @@ impl BitcoinWallet for FileBitcoinWallet {
...
-        let peg_out_output_stx = op.recipient.to_bitcoin_tx_out(op.amount);
-        let peg_out_script = Script::from_hex(&peg_out_output_stx.script_pubkey.to_hex())?;
+        let user_address_hash =
+            bitcoin::hashes::hash160::Hash::from_slice(&op.recipient.bytes()).unwrap();
+        // let peg_out_output_stx = op.recipient.to_bitcoin_tx_out(op.amount);
+        // let peg_out_script = Script::from_hex(&peg_out_output_stx.script_pubkey.to_hex())?;
+        // let peg_out_output = bitcoin::TxOut {
+        //     value: peg_out_output_stx.value,
+        //     script_pubkey: peg_out_script,
+        // };
```
Comment (netrome) [inline stacks-coordinator/src/peg_wallet.rs:108]: Nit: This should be cleaned up.

File: stacks-coordinator/src/peg_wallet.rs [hunk 2]
```diff
@@ -97,15 +98,21 @@ impl BitcoinWallet for FileBitcoinWallet {
...
-            value: peg_out_output_stx.value,
-            script_pubkey: peg_out_script,
+            value: op.amount,
+            script_pubkey: p2wpk,
         };
         Ok(bitcoin::blockdata::transaction::Transaction {
-            version: 0,
+            version: 2,
             lock_time: bitcoin::PackedLockTime(0),
             input: vec![peg_out_input],
```
Comment (netrome) [inline stacks-coordinator/src/peg_wallet.rs:117]: We should consume a UTXO from the peg wallet to fund the peg-out fulfillment. Is there a ticket to do this?

File: stacks-coordinator/src/peg_wallet.rs [hunk 3]
```diff
@@ -97,15 +98,21 @@ impl BitcoinWallet for FileBitcoinWallet {
...
             output: vec![peg_out_output],
```
Comment (netrome) [inline stacks-coordinator/src/peg_wallet.rs:118]: I guess we can't assume the input UTXOs amount to exactly what we need, so we must return the remaining balance (minus miner fee) to the peg wallet in another output.

**Review by netrome — CHANGES_REQUESTED** [review]
Cleanup request (blocking): There's some stale comments which should be cleaned up before merging.

Clarification request (not blocking): I don't see any associated issue with this PR, so it's a bit hard for me to understand the context and plan revolving this, so I have some clarifying questions about some observations on missing functionality here.

File: stacks-coordinator/src/peg_wallet.rs (hunk 2 above)
Comment (donpdonp) [reply stacks-coordinator/src/peg_wallet.rs:117]: that happens earlier in the function
```
       let peg_out_input = bitcoin::TxIn {
            previous_output: utxo,
```

Comment (donpdonp) [reply stacks-coordinator/src/peg_wallet.rs:117]: After reading some more, this needs to be reworked. PegOutRequestOp.txid is the btc transaction of the peg-out request itself.

**Comment by netrome** [comment]
Closing as this due to inactivity. The important features will find their way through @jferrant's work on the coordinator.

**Pull request closed without merging**